runtime/runtime/src/lib.rs

```diff
@@ -1319,6 +1335,10 @@ impl Runtime {
         }
 
         // And then we process the new incoming receipts. These are receipts from other shards.
+        if let Some(prefetcher) = &mut prefetcher {
+            prefetcher.clear();
+            let _queue_full = prefetcher.input_receipts(&incoming_receipts);
```

Comment:
Indeed it could and I had it this way in my first implementation.
I think it makes more sense here because we want to fetch this data after all delayed receipts have finished, if they even finish all in this block. Also, the bounded queue of requests will have more space again even if the local receipts already filled it up.
That was my thinking. But I am open to arguments to move it up if you still think it makes more sense there.